common/lanelet2_extension/include/lanelet2_extension/utility/query.h

```diff
@@ -41,6 +48,34 @@ namespace utils
 {
 namespace query
 {
+enum direction {CHECK_CHILD,CHECK_PARENT};
+struct References
+{
+    References() {};
+    struct comparator
+    {
+        template <typename PrimT>
+        bool operator()(const PrimT& prim1, const PrimT& prim2) const {
+            return prim1.id() == prim2.id();
+        }
+        bool operator()(const lanelet::RegulatoryElementConstPtr& prim1, 
+        const lanelet::RegulatoryElementConstPtr& prim2) const {
+            return prim1->id() == prim2->id();
+        }
+    };
+    std::unordered_set<lanelet::ConstLineString3d, std::hash<lanelet::ConstLineString3d>, comparator> lss;
+    std::unordered_set<lanelet::ConstLanelet, std::hash<lanelet::ConstLanelet>, comparator> llts;
+    std::unordered_set<lanelet::ConstArea, std::hash<lanelet::ConstArea>, comparator> areas;
+    std::unordered_set<lanelet::RegulatoryElementConstPtr, std::hash<lanelet::RegulatoryElementConstPtr>, comparator> regems;
+};
+/**
+ * [findReferences finds all primitives that reference the given primitive in a given map]
+ * @param  ll_Map [input lanelet map]
+ * @return        [References object with referenced element sets for each primitive layers]
+ */
```

Comment:
Add a comment noting that Polygons and Compound primitives like LaneletOrArea are not currently supported.